src/replace_landsurface_with_BARRA2R_IC.py

```diff
@@ -204,17 +205,17 @@ def swap_land_barra(mask_fullpath, ec_cb_file_fullpath, ic_date):
     
     # Read in the soil moisture data (and keep to use for replacement)
     BARRA_FIELDN = 'mrsol'
-    indir = BARRA_DIR + '/3hr/' + BARRA_FIELDN + '/v20231001'
-    barra_files = glob(indir + '/' + BARRA_FIELDN + '*' + yyyy + mm + '*nc')
-    barra_fname = indir + '/' + barra_files[0].split('/')[-1]
+    indir = os.path.join(BARRA_DIR, '3hr', BARRA_FIELDN, 'v20231001')
+    barra_files = glob(os.path.join(indir, BARRA_FIELDN + '*' + yyyy + mm + '*nc'))
+    barra_fname = os.path.join(indir, os.path.basename(barra_files[0]))
     data = get_BARRA_nc_data(barra_fname, BARRA_FIELDN, ic_date.replace('T', '').replace('Z', ''), 4, bounds)
     mrsol = data.copy()
 
     # Read in the soil temperature data (and keep to use for replacement)
     BARRA_FIELDN = 'tsl'
-    indir = BARRA_DIR + '3hr/' + BARRA_FIELDN + '/v20231001'
-    barra_files = glob(indir + '/' + BARRA_FIELDN + '*' + yyyy + mm + '*nc')
-    barra_fname = indir + '/' + barra_files[0].split('/')[-1]
+    indir = os.path.join(BARRA_DIR, '3hr', BARRA_FIELDN, 'v20231001')
+    barra_files = glob(os.path.join(indir, BARRA_FIELDN + '*' + yyyy + mm + '*nc'))
+    barra_fname = os.path.join(indir, os.path.basename(barra_files[0]))
```

Comment:
```suggestion
    barra_fname = barra_files[0]
```

See comment above about redundancy.